apple/Sources/Sargon/Drivers/EventBus/EventBus.swift

```diff
@@ -0,0 +1,67 @@
+//
+//  File.swift
+//  
+//
+//  Created by Alexander Cyon on 2024-05-05.
+//
+
+import Foundation
+import SargonUniFFI
+import AsyncExtensions
+
+// Makes it possible to type `.shared` on an initalizer/func taking
+// `some EventBusDriver` as parameter.
+extension EventBusDriver where Self == EventBus {
+	
+	/// Singleton `EventBusDriver` of type `EventBus` being an `actor`  which forwards `EventNotification`s
+	/// originally emitted by `SargonOS` (Rust side).
+	public static var shared: Self { Self.shared }
+}
+
+/// An `EventBusDriver` actor which handles incoming
+/// `EventNotifications` and forwards them to any
+/// subscriber of `notifications()`, being a multicasted
+/// async sequence.
+public final actor EventBus {
+	/// A stream we multicast on.
+	private let stream = AsyncThrowingPassthroughSubject<Element, any Error>()
+	private let subject: Subject
+#if DEBUG
+	public init() {
+		subject = .init()
+	}
+#else
+	private init() {
+		subject = .init()
+	}
+#endif
+}
+
+extension EventBus {
+	
+	public typealias Element = EventNotification
+	public typealias Subject = AsyncPassthroughSubject<Element>
+	
+	/// Singleton `EventBusDriver` of type `EventBus` being an `actor` which forwards `EventNotification`s
+	/// originally emitted by `SargonOS` (Rust side).
+	public static let shared = EventBus()
+	
+	/// A multicasted async sequence of `EventNotification` values
+	/// over time, originally emitted by `SargonOS` (Rust side).
+	public func notifications() -> AsyncMulticastSequence<EventBus.Subject, AsyncThrowingPassthroughSubject<EventBus.Element, any Error>> {
+		subject
+		 .multicast(stream)
+		 .autoconnect()
+	}
+}
+
+extension EventBus: EventBusDriver {
+	/// This method is called by `SargonOS` (Rust side) and we should
+	/// "forward" the events to subscribers (Swift swide), i.e. `@SharedReader`s of profile values,
+	/// which uses `notifications()` to subscribe to these
+	/// values.
+	public func handleEventNotification(eventNotification: EventNotification) async {
+		log.debug("Handle event: \(String(describing: eventNotification.event))")
+		subject.send(eventNotification)
```

Comment:
You mention On Rust side that `handleEventNotification` is async as to allow clients to perform some intensive work before continuing.
I guess we would await here for some kind of response for the given event?

Reply:
Glad you brought this up! I made it `handle_event_notification` be `async`  since some "subjects" in Swift at least have `async` "send" methods. e.g. `AsyncChannel` since it has backpressure semantics, not producing (sending) more values than consumer (receiver) can handle. Currently I'm not making use of that though, and I think maybe Android wont either.

A couple of days ago, after I asked everyone to review this PR, I realize we might want to have another driver later on for SargonOS to "request wallet to ask user for input", lets call this driver `OSToWalletRequestDriver` for now, **which WILL be async**, then maybe it makes sense for this to be non-async (if Android is OK with that).

The `OSToWalletRequestDriver` can be used for at least 3 things:
* We can let Sargon(OS) receive dapp requests and then Sargon(OS) tells wallet to display it, awaiting the result/output of the dapp request
* We can let Sargon(OS) perform sync jobs with Radix Network(Ledger) to check if any securitifed accounts AccessController is out of sync with a "Security Shield" and we can inform user and await input ("Re-apply shield" / "Skip")
* We can check ownership and await resolution (user clicks "Claim") instead of failing for writes when user has lost ownership of Profile

These are quite powerful things, built very easily, all necessary tool are here, just make a new driver, similar to EventBusDriver but for this purpose, and know that we display UI to user awaiting input.